buckets-mdapi/src/conditional.rs

```diff
@@ -0,0 +1,395 @@
+// Copyright 2020 Joyent, Inc.
+
+use serde_json::Value;
+use postgres::types::ToSql;
+use postgres::Transaction;
+use serde_json;
+use slog::{trace, Logger};
+use uuid::Uuid;
+use serde_derive::{Deserialize, Serialize};
+
+use crate::object;
+use crate::error;
+use crate::metrics;
+use crate::sql;
+use crate::types;
+
+#[derive(Default, Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub struct Conditions {
+    #[serde(alias = "if-match")]
+    pub if_match: Option<Vec<String>>,
+
+    #[serde(alias = "if-none-match")]
+    pub if_none_match: Option<Vec<String>>,
+
+    #[serde(alias = "if-modified-since")]
+    pub if_modified_since: Option<types::Timestamptz>,
+
+    #[serde(alias = "if-unmodified-since")]
+    pub if_unmodified_since: Option<types::Timestamptz>,
+}
+
+pub fn error(error_type: error::BucketsMdapiErrorType, msg: String) -> serde_json::Value {
```

Comment:
Yea good point. I've changed this to be an internal function now that always returns a PreconditionFailedError. It's a bit thin on the ground with the changes to BucketsMdapiError as well, and I'm not totally sure on its utility any more. Thinking about it, it could probably just do with being removed